### PR TITLE
Refactoring Tests for User model

### PR DESCRIPTION
**Pull request by lirantal**
re-factoring the server model tests to create user model from schema only during tests, so that we can re-use mongodb's _v versioning fields, and also cleaning up user tests on each test iteration

### CODE REVIEW

File: modules/users/tests/server/user.server.model.tests.js [hunk 1]
```diff
@@ -57,59 +57,90 @@ describe('User Model Unit Tests:', function () {
...
+      var _user2 = new User(user2);
+      _user.save(function () {
+        _user2.save(function (err) {
           should.exist(err);
-          done();
+          _user.remove(function (err) {
+            should.not.exist(err);
+            done();
+          });
+
```
Comment (ilanbiala) [inline modules/users/tests/server/user.server.model.tests.js]: nitpick, but inconsistent space? If we want more spacing, we should have above and below.

File: modules/users/tests/server/user.server.model.tests.js [hunk 2]
```diff
@@ -57,59 +57,90 @@ describe('User Model Unit Tests:', function () {
...
     });
 
     it('should confirm that saving user model doesnt change the password', function (done) {
-      user.firstName = 'test';
-      var passwordBefore = user.password;
-      return user.save(function (err) {
-        var passwordAfter = user.password;
-        passwordBefore.should.equal(passwordAfter);
-        done();
+
```
Comment (ilanbiala) [inline modules/users/tests/server/user.server.model.tests.js]: Spacing

File: modules/users/tests/server/user.server.model.tests.js [hunk 3]
```diff
@@ -57,59 +57,90 @@ describe('User Model Unit Tests:', function () {
...
+            should.not.exist(err);
+            done();
+          });
+        });
       });
     });
 
     it('should be able to save 2 different users', function (done) {
-      user.remove(function (err) {
+
```
Comment (ilanbiala) [inline modules/users/tests/server/user.server.model.tests.js]: spacing

**Comment by ilanbiala** [comment]
Looks good, just some minor nitpicks.

**Comment by lirantal** [comment]
Thanks @ilanbiala , missed your CR :)
I addressed the spacing but I updated the convention there to have a line space between var declares and the rest of the code for readability as otherwise it just looks a lot of code mangled together.

**Comment by ilanbiala** [comment]
That looks awesome to me!

**Comment by lirantal** [comment]
Cool, thanks.

**Comment by codydaig** [comment]
LGTM

**Pull request merged**